### PR TITLE
Bump ktlint, Checkstyle, Java version

### PR DESCRIPTION
**Pull request by TacoTheDank**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [x] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Move plugins to the DSL block because it's cool
- Compile with Java 11 (we can because we compile on SDK 30)
- Update Sonarqube 3.1.1 -> 3.3 ([changelog](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-gradle/))
- Update ktlint 0.40.0 -> 0.43.2 (and run it) ([changelog](https://github.com/pinterest/ktlint/releases))
- Update Checkstyle 8.38 -> 9.2 (and run it) ([changelog](https://checkstyle.sourceforge.io/releasenotes.html))

#### APK testing 
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/local/feed/service/FeedLoadService.kt [hunk 1]
```diff
@@ -183,28 +183,23 @@ class FeedLoadService : Service() {
 
         subscriptions
             .take(1)
-
```
Comment (litetex) [inline app/src/main/java/org/schabi/newpipe/local/feed/service/FeedLoadService.kt:186]: Why did you remove the lines? Or did a plugin do it?

**Review by litetex — COMMENTED** [review]
LGTM some minor things

File: app/src/main/java/org/schabi/newpipe/local/feed/service/FeedLoadService.kt (hunk 1 above)
Comment (TacoTheDank) [reply app/src/main/java/org/schabi/newpipe/local/feed/service/FeedLoadService.kt:186]: This was ktlint going off from new lint additions in newer versions. Doing formatKtlint did this.

Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/local/feed/service/FeedLoadService.kt:186]: Not a problem, I think

**Review by litetex — APPROVED** [review]
